pkg/controller/master/backup/restore.go

```diff
@@ -209,6 +230,38 @@ func (h *RestoreHandler) RestoreOnRemove(key string, restore *harvesterv1.Virtua
 	return nil, nil
 }
 
+// pv naming convention from externel-provisioner
+// porting from https://github.com/kubernetes-csi/external-provisioner/blob/90eae32d3a7352590500073b72b9a07f43adc881/pkg/controller/controller.go#L420-L436
+func makeVolumeName(prefix, pvcUID string, volumeNameUUIDLength int) (string, error) {
+	// create persistent name based on a volumeNamePrefix and volumeNameUUIDLength
+	// of PVC's UID
+	if len(prefix) == 0 {
+		return "", fmt.Errorf("Volume name prefix cannot be of length 0")
+	}
+	if len(pvcUID) == 0 {
+		return "", fmt.Errorf("corrupted PVC object, it is missing UID")
+	}
+	if volumeNameUUIDLength == -1 {
+		// Default behavior is to not truncate or remove dashes
+		return fmt.Sprintf("%s-%s", prefix, pvcUID), nil
+	}
+	// Else we remove all dashes from UUID and truncate to volumeNameUUIDLength
+	return fmt.Sprintf("%s-%s", prefix, strings.Replace(string(pvcUID), "-", "", -1)[0:volumeNameUUIDLength]), nil
```

Comment:
iirc, you can leave off the 0 if you prefer.

```suggestion
	return fmt.Sprintf("%s-%s", prefix, strings.Replace(string(pvcUID), "-", "", -1)[:volumeNameUUIDLength]), nil
```

Reply:
This is porting from `externel-provisioner`,  I'd like to keep it as where it's from, thanks.